### PR TITLE
Nightly PyTorch version is now 2.0

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

PyTorch nightlies have 2.0 as the version number, signaling that there won't be a 1.14 release. https://github.com/pytorch/pytorch/pull/90491

### Does your PR introduce any breaking changes? If yes, please list them.

None

cc @borda @justusschock @awaelchli

### CODE REVIEW

**Review by justusschock — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Comment by Borda** [comment]
> signaling that there won't be a 1.14 release

can you share a reference to why you thin so?

File: src/pytorch_lightning/callbacks/quantization.py [hunk 1]
```diff
@@ -26,8 +26,8 @@
 from torch.quantization import FakeQuantizeBase
 
 import pytorch_lightning as pl
+from lightning_lite.utilities.imports import _TORCH_GREATER_EQUAL_1_11, _TORCH_GREATER_EQUAL_1_12
```
Comment (Borda) [inline src/pytorch_lightning/callbacks/quantization.py:29]: can we do this in separate PR?

File: src/pytorch_lightning/utilities/imports.py [hunk 2]
```diff
@@ -13,20 +13,11 @@
...
-import platform
-import sys
 
 import torch
 from lightning_utilities.core.imports import compare_version, module_available, package_available, RequirementCache
 
-_IS_WINDOWS = platform.system() == "Windows"
-_PYTHON_GREATER_EQUAL_3_8_0 = (sys.version_info.major, sys.version_info.minor) >= (3, 8)
-_PYTHON_GREATER_EQUAL_3_10_0 = (sys.version_info.major, sys.version_info.minor) >= (3, 10)
 _TORCH_LESSER_EQUAL_1_10_2 = compare_version("torch", operator.le, "1.10.2")
```
Comment (Borda) [inline src/pytorch_lightning/utilities/imports.py:20]: why this stay and others are moved?

File: src/lightning_lite/accelerators/cuda.py [hunk 3]
```diff
@@ -20,7 +20,7 @@
 import torch
 
 from lightning_lite.accelerators.accelerator import Accelerator
-from lightning_lite.utilities.imports import _TORCH_GREATER_EQUAL_1_13, _TORCH_GREATER_EQUAL_1_14
+from lightning_lite.utilities.imports import _TORCH_GREATER_EQUAL_1_13, _TORCH_GREATER_EQUAL_2_0
```
Comment (Borda) [inline src/lightning_lite/accelerators/cuda.py:23]: I would still keep this...

**Review by Borda — COMMENTED** [review]
seem that the PR title is misleading as it is rather using imports from Lite then changing anything to 3.0

File: src/lightning_lite/accelerators/cuda.py (hunk 3 above)
Comment (carmocca) [reply src/lightning_lite/accelerators/cuda.py:23]: There's no 1.14: https://github.com/pytorch/pytorch/pull/90491

File: src/pytorch_lightning/utilities/imports.py (hunk 2 above)
Comment (carmocca) [reply src/pytorch_lightning/utilities/imports.py:20]: Lite does not define this one. The others are, so they were duplicated

File: src/pytorch_lightning/callbacks/quantization.py (hunk 1 above)
Comment (carmocca) [reply src/pytorch_lightning/callbacks/quantization.py:29]: It's just removing one duplicated variable

File: src/lightning_lite/accelerators/cuda.py (hunk 3 above)
Comment (Borda) [reply src/lightning_lite/accelerators/cuda.py:23]: it was nor merged, so not sure :confused:

File: src/pytorch_lightning/callbacks/quantization.py (hunk 1 above)
Comment (Borda) [reply src/pytorch_lightning/callbacks/quantization.py:29]: hmm but your removing is 95% of this PR :P

File: src/lightning_lite/accelerators/cuda.py (hunk 3 above)
Comment (carmocca) [reply src/lightning_lite/accelerators/cuda.py:23]: It is merged. See the label in the PR

File: src/pytorch_lightning/callbacks/quantization.py (hunk 1 above)
Comment (Borda) [reply src/pytorch_lightning/callbacks/quantization.py:29]: I would just update the PR name, but @carmocca I know may complain :)
